src/vc_util/src/lib.rs

```diff
@@ -1007,4 +1056,165 @@ mod tests {
+        CredentialSpec {
+            credential_type: "vcWithOneArg".to_string(),
+            arguments: Some(args),
+        }
+    }
+
+    fn credential_specs_for_test() -> Vec<CredentialSpec> {
+        vec![
+            credential_spec_with_0_args(),
+            credential_spec_with_1_arg(),
+            credential_spec_with_2_args(),
+        ]
+    }
+
+    fn vc_claims_for_spec(spec: &CredentialSpec) -> Map<String, Value> {
+        let mut claims = Map::new();
+        let types = vec![
+            Value::String("VerifiableCredential".to_string()),
+            Value::String(spec.credential_type.to_string()),
+        ];
+        claims.insert("type".to_string(), Value::Array(types));
+        let mut arguments = Map::new();
+        if let Some(args) = spec.arguments.as_ref() {
+            for arg in args {
+                arguments.insert(arg.0.clone(), arg.1.clone().into());
+            }
+        }
+        let mut subject = Map::new();
+        subject.insert(spec.credential_type.clone(), Value::Object(arguments));
+        claims.insert("credentialSubject".to_string(), Value::Object(subject));
+        claims
+    }
+
+    #[test]
+    fn should_validate_claims_match_spec() {
+        for spec in credential_specs_for_test() {
+            let claims = vc_claims_for_spec(&spec);
+            validate_claims_match_spec(&claims, &spec)
+                .unwrap_or_else(|_| panic!("failed for spec: {:?}", spec));
+        }
+    }
+
+    #[test]
+    fn should_fail_validate_claims_match_spec_if_wrong_type() {
+        for spec in credential_specs_for_test() {
+            // Construct claims with wrong "type" entry.
+            let mut claims = vc_claims_for_spec(&spec);
+            claims.insert(
+                "type".to_string(),
+                Value::Array(vec![Value::String("WrongType".to_string())]),
+            );
+            let result = validate_claims_match_spec(&claims, &spec);
+            assert_matches!(result, Err(e) if format!("{:?}", e).contains("missing credential_type in type-claim"));
+        }
+    }
+
+    #[test]
+    fn should_fail_validate_claims_match_spec_if_missing_credential_type_claim() {
+        for spec in credential_specs_for_test() {
+            // Construct claims without "credential_type"-claim.
+            let mut claims = vc_claims_for_spec(&spec);
+            claims
+                .get_mut("credentialSubject")
+                .expect("missing credentialSubject")
+                .as_object_mut()
+                .expect("wrong credentialSubject")
+                .remove(&spec.credential_type)
+                .expect("missing credential_type claim");
+            let result = validate_claims_match_spec(&claims, &spec);
+            assert_matches!(result, Err(e) if format!("{:?}", e).contains("missing credential_type claim"));
+        }
+    }
+
+    #[test]
+    fn should_fail_validate_claims_match_spec_with_extra_args_in_credential_type_claim() {
```

Comment:
Maybe for better readability it would be ok to not repeate the function name verbatim and just use `should_fail_validation_...`?

Reply:
With repeated function name it is clear what is being tested, and also I find it useful for filtering the tests.  
I don't feel strongly about it though, so if it is only useful for me, I can rename (leaving for now, can do another PR)